Sources/core/objects/Gun.cpp

```diff
@@ -0,0 +1,67 @@
+#include "Gun.h"
+
+Gun::Gun(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+
+}
+
+bool Gun::Handle_Ammo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Gun::Shoot(VDir target,id_ptr_on<Human> shooter)
+{	
+    if(Handle_Ammo())
+    {
+      if(GetOwner()->GetOwner().valid())
+      {
+          id_ptr_on<Projectile> p = Create<Projectile>(Bullet::T_ITEM_S(),GetOwner()->GetOwner());
+          p->MakeMovementPattern(target,shooter);
+      }
+             //       Create<Item>(Bullet_Casing::T_ITEM_S(),GetOwner);	
+    }
+    else
+    {
+          PlaySoundIfVisible("empty.ogg",owner.ret_id());
+		// the *click* text from ss13
+    }	
+}
+
+int Gun::AddAmmunition()
+{
+    if (ammunition_ < max_ammunition_)
+    {
+        ++ammunition_;
+	return ammunition_;
+    }    
+    if (ammunition_ > max_ammunition_)
+    {
+	--ammunition_;
+        return ammunition_;
+    }
```

Comment:
And what will happen when `ammunition_` == `max_ammunition_`? It is really huge mistake, and maybe it causes unsyncs. This function a little bit weird. It should probably return `true` or `false` (was addition successfull or not), and should cover all cases. Also it is better to rename it, because everywhere else functions which work with ammution are called `<Something>Ammo`.
```
bool Gun::AddAmmo()
{
    if (amunition_ < max_amunition_)
    {
        ++ammunition_;
        return true;
    }
    return false;
}
```